experimental/inductor.py

```diff
@@ -0,0 +1,218 @@
+import torch
+from torch import nn
+from torch.fx.experimental.proxy_tensor import make_fx
+from torch._functorch.aot_autograd import run_functionalized_fw_and_collect_metadata, aot_function
+import functorch
+from functorch.compile import min_cut_rematerialization_partition
+
+from torch._functorch.eager_transforms import functionalize
+
+from torch import autograd
+from torch._inductor.compile_fx import compile_fx_inner
+
+
+from torch._inductor.decomposition import select_decomp_table
+inductor_decomps = select_decomp_table()
+
+
+
+from copy import copy
+all_decomps = copy(inductor_decomps)
+
+
+def prepare_args(attr_vals, unused_inputs_idx, args):
+    """
+    modify args to pass into a function compiled with make_functional_fx
+    args: original arguments
+    unused_inputs_idx, attr_vals: data returned from make_functional_fx
+    """
+    all_args = list(args) + list(attr_vals)
+    return [a for i, a in enumerate(all_args) if i not in unused_inputs_idx]
+
+
+def make_functional_fx(fn):
+    """
+    Given a function, does make_fx(functionalize(make_fx(fn))).
+    Some input-related transformations need to be done before functionalize.
+    Returns an GraphModule to be fed to inductor.
+    """
+    def call(*args, **kwargs):
+        # 1. call make_fx for the first time
+        k1 = make_fx(fn)
+        k2 = k1(*args, **kwargs)
+
+
+        # 2. Turn getattr into inputs
+        #    - This is needed for functionalize to work. Functionalize expects
+        #      all tensors to be passed as inputs to the function being functionalized.
+        #    - We collect all concrete getattr inputs into "attr_vals"
+        from collections import defaultdict
+        attr_vals_map = dict()
+        attr_nodes = defaultdict(lambda: [])
+
+        for p in k2.graph.nodes:
+            if p.op == 'get_attr':
+                val = getattr(k2, p.target)
+                attr_vals_map[p.target] = val
+                attr_nodes[p.target].append(p)
+
+        after_last_placeholder = None
```

Comment:
nit - weird name/logic? - this is the last arg + 1 right? logically, you are doing "insert before the first argument after the last placeholder" which is a long way of saying "insert after the last arg" - maybe clean this up to just insert after the last placeholder?

Reply:
yeah i want to insert a new placeholder after the last placeholder, which means we'll be appending args to the input of the graph . i'll change to inserting_after.